### ISSUE TITLE
Feature request: toggle Ecosia as new tab page on or off

### ISSUE THREAD

**Issue by Gersonzao** [aidanhorton/ecosia-darkmode-extensions#46]
I prefer my browser's default new tab page and I just want an extension that doesn't have Ecosia as a new tab page

**Comment by OppnedKatt** [comment]
When installing the exstension, click the pop-up button that says something on the lines of "revert changes". Or maybe it appears when you first go to a new tab. Some people wanted a dark mode version of the ecosia new tab page, so that's why it's here. Because of how new tab pages work, we can't just add a toggle for the new tab page. You need to revert the changes on installation or for time opening a new tab after installation.

**Comment by OppnedKatt** [comment]
You can delete the extension, then reinstall it, and revert the changes when prompted with it to remove the ecosia new tab page that comes from this exstension.

**Comment by Gersonzao** [comment]
When I do it, it disables the extension

**Comment by aidanhorton** [comment]
Yeah as Oppned says, its impossible to add an option for this in the extension due to how the Chrome Extension API works.

It's weird, the button as you mention does just seem to disable the extension - you would've thought Chrome would be smart enough to handle this!
![image](https://user-images.githubusercontent.com/22588585/179533041-d289655b-19c0-48c1-bbdb-57a51fb79387.png)

I'll investigate this and see if anything can be done!

**Comment by aidanhorton** [comment]
Update, it may not be impossible to have it optional, will confirm when I know more info :)

**Comment by aidanhorton** [comment]
While investigating this further (it may be possible to add an option), a workaround for now is you could always install an extension after the Ecosia darkmode one which overrides the newtab page with one that you like - there may even be one that overrides it back to the default style!

**Comment by aidanhorton** [comment]
So I've actually found a way to add an option in the extension for this - they must've changed this recently because I looked at adding this option a year back but had no luck.

Hopefully the new version with this fix will be uploaded sometime within the next 3 days!

**Comment by OppnedKatt** [comment]
I still wish the chrome devs would add a section in their settings for selecting which extension's new tab page you want to use, if you want to use any, even though I don't use Chrome myself.

**Comment by aidanhorton** [comment]
> I still wish the chrome devs would add a section in their settings for selecting which extension's new tab page you want to use, if you want to use any, even though I don't use Chrome myself.

It really doesn't make sense does it. FireFox has this option so the fact that Chrome doesn't is pretty shocking!


@Gersonzao you'll be happy to know that v1.4.2.0 is currently being reviewed by the extension store, it should be approved in around 3 hours from this message, so you'll have an option in the extension settings (found by clicking the extension icon), to turn off the newtab override! Let me know if you have any issues, I'll close this issue in a few days or upon confirmation of the fix working!

**Comment by OppnedKatt** [comment]
I downloaded the extension on chrome, and this seems to work.

**Comment by Chirunos** [comment]
I'm very sorry for the big delay. When I disable the option and I open a new tab, I can see the Ecosia tab **very** quickly and then I get redirected to `chrome-search://local-ntp/local-ntp.html`. I'm using Brave Browser 1.46.70 Beta

**Comment by aidanhorton** [comment]
> I'm very sorry for the big delay. When I disable the option and I open a new tab, I can see the Ecosia tab **very** quickly and then I get redirected to `chrome-search://local-ntp/local-ntp.html`. I'm using Brave Browser 1.46.70 Beta


@Gersonzao Thanks for reporting this! I'll do a bit of testing, but does Brave have the option to decide which newtab to use in browser? (I'll check this soon so no worries if you don't get back to me on this).

Maybe I'll just have to simply upload a different fix to the brave extension store (assuming there is one!)

**Comment by Chirunos** [comment]
Brave Browser doesn't have its own extension store, but I think there's a very low chance that it's ever going to happen:
https://github.com/brave-experiments/brave-extension-store
https://community.brave.com/t/feature-un-request-extensions-store/50096

AFAIK, there's no way you can change the new tab (in a setting that you can change precisely of which extension you want to use). When I just install the extension or when I disable and enable it and I open a new tab, it asks me if I want to use the new tab set by the extension or not. When I say I don't want to, it simply disables the extension. The settings for the new tab can be found here: `brave://settings/newTab`
![image](https://user-images.githubusercontent.com/85644671/197403904-af41a765-0d54-4aef-bbba-102039833e54.png)

**Comment by aidanhorton** [comment]
@Gersonzao if there is no extension store (meaning I can't upload a Brave version of the extension), then it leaves me in a tough spot. I can't disable this feature as it seems to work on Chrome, and it was a fairly high requested feature (better than no option anyways!). It was hard enough finding a single way to make the 'Override new tab' setting work, and as far as I know, the redirect method is the only way of actually accomplishing this.
I'll do a bit of research, but it's looking like the options for fixing this are very limited unfortunately! Let me know how much of a problem it is for you.

**Comment by Chirunos** [comment]
This isn't a problem for me anymore. I have made a style myself and I only use the extension for testing. I think you could make 2 switches, 1 that works for most Chromium-based browsers and another one for Brave Browser, but I think that would make a bad UX... or make the extension automatically detect the browser
